Services/TagService.cs

```diff
@@ -0,0 +1,37 @@
+using Neo4jClient;
+using TAG.Nodes;
+using TAG.Extensions;
+using TAG.Constants;
+using TAG.DTOS;
+using TAG.Services.Interfaces;
+
+namespace TAG.Services
+{
+    public class TagService : ITagService
+    {
+        private readonly IGraphClient _graphClient;
+
+        public TagService(IGraphClient graphClient)
+        {
+            _graphClient = graphClient;
+        }
+
+        public async Task<TagSearchResponseDTO> SearchTagsAsync(TagSearchRequestDTO request)
+        {
+            var query = _graphClient.Cypher
+                .Match("(tag:Tag)")
+                .WhereIf<TagNode>(request.Name != null && request.Name != "", (tag) => tag.Type.Contains(request.Name));
```

Comment:
You can simplify this check using string.IsNullOrEmpty(request.Name) or even string.IsNullOrWhiteSpace(request.Name)

Reply:
fixed